### PR TITLE
Add OpenMP and Makefile

### PR DESCRIPTION
**Pull request by MatthewKhouzam**
Makes two executables to compare.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Variables
+CC = gcc
+CFLAGS = -Wall
+OPTIMIZATON1 = -o1
```
Comment (marcdumais-work) [inline Makefile]: needs to be uppercase "O"s

Comment (marcdumais-work) [reply Makefile]: By far the biggest difference in exec time is between no optimization (O0) and some optimization (O1). However I did not notice much interesting between the two traces, so not sure if we should have O0?

Comment (marcdumais-work) [reply Makefile]: O3 does not change much for either. WDYT having from O0 to O2 for both? Too much?

Comment (marcdumais-work) [reply Makefile]: I pushed a commit on top, for the comments above.

**Review by marcdumais-work — APPROVED** [review]
LGTM! Please have a look at the commit I added before merging!

**Review by MatthewKhouzam — COMMENTED** [review]
LGTM!

**Pull request merged**